R/Renderer.R

```diff
@@ -116,23 +138,32 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
         block_content
       )
     },
-    rcodeBlock2md = function(block) {
+    rcodeBlock2md = function(block, report_type) {
       params <- block$get_params()
       params <- lapply(params, function(l) if (is.character(l)) shQuote(l) else l)
       block_content <- block$get_content()
-      paste(
-        sep = "\n",
-        collapse = "\n",
-        "### ",
+      if (identical(report_type, "powerpoint_presentation")) {
+        block_content_lst <- split_text_into_blocks(block_content, 30)
+        paste(
+          unlist(
+            lapply(seq_along(block_content_lst), function(b) {
+              sprintf(
+                "### \n\n```{r, echo=FALSE}\ncode_block(data.frame(\n%s))\n```\n",
+                shQuote(block_content_lst[[b]])
+              )
+            })
+          ),
+          collapse = "\n\n"
+        )
```

Comment:
1. don't `lapply(seq_along(<mylist>), function(i) mylist[[i]])`, do `lapply(<mylist>, function(x) x)`
2. `sprintf` is vectorized, there is no need for `lapply`.

Reply:
Thank you. I have updated it.